min-turnup/phase1/azure/lib/azure.jsonnet

```diff
@@ -0,0 +1,285 @@
+function(cfg1)
+	local cfg = cfg1.phase1;
+	local master_private_ip = cfg.azure.master_private_ip;
+	local names = {
+		resource_group: "%(instance_prefix)s" % cfg,
+		master_public_ip: "%(instance_prefix)s-master-pip" % cfg,
+		availability_set: "%(instance_prefix)s-as" % cfg,
+		storage_account: "${replace(\"%(instance_prefix)sstrg\", \"-\", \"\")}" % cfg,
+		storage_container: "kube0000%(instance_prefix)s" % cfg,
+		vnet: "%(instance_prefix)s-vnet" % cfg,
+		subnet: "%(instance_prefix)s-subnet" % cfg,
+		route_table: "%(instance_prefix)s" % cfg,
+		security_group: "%(instance_prefix)s-nsg" % cfg,
+		master_nic: "%(instance_prefix)s-master-nic" % cfg,
+		master_vm: "%(instance_prefix)s-master" % cfg,
+		node_nic: "%(instance_prefix)s-node-nic" % cfg,
+		node_vm: "%(instance_prefix)s-node" % cfg,
+	};
+	{
+		variable: {
+			subscription_id: { default: cfg.azure.subscription_id },
+			tenant_id: { default: cfg.azure.tenant_id },
+			client_id: { default: cfg.azure.client_id },
+			client_secret: { default: cfg.azure.client_secret },
+		},
+		provider: {
+			azurerm: {
+				subscription_id: "${var.subscription_id}",
+				tenant_id: "${var.tenant_id}",
+				client_id: "${var.client_id}",
+				client_secret: "${var.client_secret}",
+			}
+		},
+		resource: {
+			azurerm_resource_group: {
+				rg: {
```

Comment:
True, but doing 'a_thing: { the_thing: { ... } }' doesn't look too much better. Is there a jsonnet trick for such things?